diff --git a/node_modules/prerender-redis-cache/lib/prerenderRedisCache.js b/node_modules/prerender-redis-cache/lib/prerenderRedisCache.js
index 14d7f23..b5bd2a4 100644
--- a/node_modules/prerender-redis-cache/lib/prerenderRedisCache.js
+++ b/node_modules/prerender-redis-cache/lib/prerenderRedisCache.js
@@ -12,10 +12,20 @@ var REDIS_URL = process.env.REDISTOGO_URL ||
 var url = require('url');
 var TTL = process.env.PAGE_TTL || 86400;
 
+let options = {};
+let connectionPassword;
+
 // Parse out the connection vars from the env string.
 var connection = url.parse(REDIS_URL);
+
+// Parse out password from the connection string
+if (connection.auth) {
+    connectionPassword = connection.auth.split(':')[1];
+    options.password = connectionPassword
+}
+
 var redis = require('redis');
-var client = redis.createClient(connection.port, connection.hostname);
+var client = redis.createClient(connection.port, connection.hostname, options);
 var redisOnline = false;
 
 var STATUS_CODES_TO_CACHE = {
@@ -38,9 +48,8 @@ connection.path = (connection.pathname || '/').slice(1);
 connection.database = connection.path.length ? connection.path : '0';
 client.select(connection.database);
 
-// Parse out password from the connection string
 if (connection.auth) {
-    client.auth(connection.auth.split(':')[1]);
+    client.auth(connectionPassword);
 }
 
 // Catch all error handler. If redis breaks for any reason it will be reported here.
@@ -110,3 +119,5 @@ module.exports = {
         next();
     }
 };
+
+
